### PR TITLE
properly integrate add/modify flight into ViewModel

### LINKED ISSUE #196: integrate ViewModel into add flight

### LINKED ISSUE #198: integrate ViewModel into ModifyFlight

### LINKED ISSUE #199: unify FlightModel with viewmodel from calendars

### PR DESCRIPTION
**Pull request by RAFInery**
closes #196
closes #198 
closes #199

- add and modify flight are now entirely connected to VM and DB
- connected confirm flight screen i.e. screen is reachable but not yet connected to DB and the VM
- updated `FlightsViewModel` to fetch the flight by a flightId through a `StateFlow`
- small updates of tests for the new asynchronous architecture
- added loading component

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/navigation/HomeGraph.kt [hunk 1]
```diff
@@ -34,76 +29,42 @@ fun NavGraphBuilder.homeGraph(
...
     composable(Route.HOME) {
-      val flightsViewModel =
-          FlightsViewModel.createViewModel(
-              flightTable = repository.flightTable,
-              balloonTable = repository.balloonTable,
-              basketTable = repository.basketTable,
-              flightTypeTable = repository.flightTypeTable,
-              vehicleTable = repository.vehicleTable)
+      val flightsViewModel = FlightsViewModel.createViewModel(repository)
+      flightsViewModel.refresh()
```
Comment (Baribal02) [inline app/src/main/java/ch/epfl/skysync/navigation/HomeGraph.kt:34]: flightsViewModel should be a shared view model

**Review by Baribal02 — APPROVED** [review]
Overall looks good to me

File: app/src/main/java/ch/epfl/skysync/navigation/MainGraph.kt [hunk 2]
```diff
@@ -17,10 +17,8 @@ fun MainGraph(
...
     uid: String?
 ) {
-  NavHost(
-      navController = navHostController,
-      startDestination = if (uid == null) Route.LOGIN else Route.MAIN) {
-        homeGraph(repository, navHostController, uid)
-        composable(Route.LOGIN) { LoginScreen(signInLauncher = signInLauncher) }
-      }
+  NavHost(navController = navHostController, startDestination = Route.MAIN) {
+    homeGraph(repository, navHostController, uid)
```
[the remark this answers is not in the document]
Comment (RAFInery) [reply app/src/main/java/ch/epfl/skysync/navigation/MainGraph.kt]: thanks. I've disabled it for testing but now it is switched back to login screen.

File: app/src/main/java/ch/epfl/skysync/navigation/HomeGraph.kt (hunk 1 above)
Comment (RAFInery) [reply app/src/main/java/ch/epfl/skysync/navigation/HomeGraph.kt:34]: For now we don't really need it but maybe at some point we can switch to a shared model. This requires some restructuring of the nav graph though

**Pull request merged**